db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -3957,12 +3957,17 @@ void DBImpl::GetSnapshotContext(
   *snapshot_seqs = snapshots_.GetAll(earliest_write_conflict_snapshot);
 }
 
-Status DBImpl::WaitForCompact(bool wait_unscheduled) {
-  // Wait until the compaction completes
+Status DBImpl::WaitForCompact(bool abort_on_pause) {
   InstrumentedMutexLock l(&mutex_);
+  if (shutting_down_.load(std::memory_order_acquire)) {
```

Comment:
These need to be checked on every iteration through the loop.

Reply:
Ah got it. Makes sense. QQ, though. If there's nothing to wait but the db is shutting down, should we expect ShutdownInProgress() or OK()?

Reply:
Per discussion over chat, we will move on with returning ShutdownInProgress() regardless.